src/main/java/com/artipie/management/Users.java

```diff
@@ -347,7 +170,8 @@ public boolean equals(final Object other) {
                 final User user = (User) other;
                 res = Objects.equals(this.uname, user.uname)
                     && Objects.equals(this.mail, user.mail)
-                    && Objects.equals(this.ugroups, user.ugroups);
+                    && this.ugroups.size() == user.ugroups.size()
+                    && this.ugroups.containsAll(user.ugroups);
```

Comment:
@olenagerasimova it seems that you forgot to revert changes in `Users.equals()` method